### DISCUSSION TITLE
Run ensemble of storms

### DISCUSSION THREAD

**Discussion by angejake11** [pyswmm/pyswmm#527] [General]
Hi all, 

I was wondering if it is possible to use pyswmm to run different 10 different rainfall files, corresponding to 10 different temporal patterns, and if so how would one go about doing this?

**Comment by dickinsonre** [comment]
isn't this purpose a primary task  of PySWMM ,

**Reply by angejake11** [reply]
It probably is. I have absolutely no prior experience with pyswmm or even coding at all for that matter so was unsure how I would go about incorporating this into a script?

**Comment by bemcdonnell** [comment]
@angejake11, this is definitely something pyswmm is good at doing. Can you map out your intended workflow so we can learn about what you’re trying to do? I’m wondering how your are building your rainfall data into a time series, in particular.

**Comment by angejake11** [comment]
Hi @bemcdonnell, thank you for getting back to me. I'm wanting to explore what impact climate change will have on the effectiveness of a detention basin at reducing peak flow rates and then potentially how smart control can be utilized to mitigate these increases.
 At present I have a simple input file setup for a lot size catchment (Attached), and I have an excel spreadsheet setup with macros that is capable of generating .txt files for a single storm event (example attached). I have three different climate scenarios I want to consider (present, 2100 medium severity case, 2100 high severity case) for which I have the capability to generate these rainfall .txt files already.
 
My number one issue at the moment is trying to figure out a way to be able to loop through multiple txt files for a particular event (frequency and duration), and determine the mean and median runoff,  without having to individually run 10 different temporal patterns if that makes sense?

I have also attached a copy of script that I am currently using and hoping to adapt.

Thank you!

[SWMM_Passive.zip](https://github.com/pyswmm/pyswmm/files/15314712/SWMM_Passive.zip)
